crates/emitter/src/forward_jump_emitter.rs

```diff
@@ -0,0 +1,73 @@
+use super::emitter::{BytecodeOffset, EmitError};
+use crate::ast_emitter::AstEmitter;
+
+#[derive(Debug)]
+pub enum JumpKind {
+    Coalesce,
+    LogicalAnd,
+    LogicalOr,
+    IfEq,
+    Goto,
+}
+
+#[derive(Debug)]
+#[must_use]
+pub struct JumpPatchEmitter {
+    pub offset: BytecodeOffset,
+}
+impl JumpPatchEmitter {
+    pub fn emit(self, emitter: &mut AstEmitter) -> Result<(), EmitError> {
+        emitter.emit.patch_jump_target(vec![self.offset]);
+        emitter.emit.jump_target();
+        Ok(())
+    }
+}
+
+impl Drop for JumpPatchEmitter {
+    fn drop(&mut self) {
+        panic!("Dropped JumpPatchEmitter without calling it");
+    }
+}
+
+// Struct for emitting bytecode for forward jump.
+#[derive(Debug)]
+pub struct ForwardJumpEmitter {
+    pub jump: JumpKind,
+}
+impl ForwardJumpEmitter {
+    pub fn emit(&mut self, emitter: &mut AstEmitter) -> Result<JumpPatchEmitter, EmitError> {
+        let offset: BytecodeOffset = emitter.emit.bytecode_offset();
+        self.emit_jump(emitter)?;
+
+        // The JITs rely on a jump target being emitted after the
+        // conditional jump
+        emitter.emit.jump_target();
```

Comment:
https://searchfox.org/mozilla-central/rev/49ed791eec93335abfe6c2880f84c324e73e47e6/js/src/frontend/BytecodeEmitter.cpp#401-412
https://searchfox.org/mozilla-central/rev/49ed791eec93335abfe6c2880f84c324e73e47e6/js/src/vm/BytecodeUtil.h#376-393

`JumpTarget` is necessary only for conditional jump (when `BytecodeFallsThrough` returns `true` above),
so, this shouldn't be emitted for `JumpKind::Goto`.
Can you add similar function and make this conditionally emit `JumpTarget` ?